lib/logreader.c

```diff
@@ -285,7 +292,7 @@ static void
 log_reader_suspend_until_awoken(LogReader *self)
 {
   self->immediate_check = FALSE;
-  poll_events_suspend_watches(self->poll_events);
+  log_reader_disable_watches(self);
```

Comment:
Disabling the `idle_timer` during *suspend* does make things awkward, as the `LogReader` and the `LogProtoServer` has an interface that allow the proto to set the *timeout* value for `idle_timer` (`log_proto_server_prepare`). Where it has to specify a `LogProtoPrepareAction` and can change the *timeout*.

From the proto point of view it makes sense to ask the `LogReader` not to do anything (`LPPA_SUSPEND`), and set an amount of time to keep the "connection".
This breaks any proto that may want to set `idle_timeout` now.

Reply:
It is an interesting question whether we want the idle timer running when a proto suspends the source with `LPPA_SUSPEND`.

Other callers of `log_reader_suspend_until_awoken()` are interesting too. When a destination (with flow control) suspends the source, it seems intuitive to stop the timer, since we want to close _idle_ connections, and a suspended connection is obviously not idle (it was actually active and faster than the destination, this triggered the suspension).

However, we can't detect a closed or broken connection when the source is suspended, so a timeout could actually help this scenario. There are other ways to do something with closed/broken connections, I just wanted to mention this.

Reply:
We just had a discussion with @szemere about this topic from `LogProtoServer` point of view.
My argument was, that why would we need to care if a connections is broken on a suspended connection?
Clients would try to re-connect to our server which was just suspended, probably making a new connection, and eating up another window size from the queue.

Also, the broken link can be re-established while it is suspended, which would make the error transparent, right?

With all that, I think we can stop the idle timer when a source is suspended.

Reply:
>We discussed the open question in the team meeting, double checked the code, and decided to stick with the current version.